drivers/hih6130/hih6130.c

```diff
@@ -0,0 +1,189 @@
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    i2c_release(dev->i2c);
+
+    return 0;
+}
+
+int hih6130_init(hih6130_t *dev, i2c_t i2c, uint8_t address)
+{
+    /* write device descriptor */
+    dev->i2c = i2c;
+    dev->addr = address;
+    return 0;
+}
+
+static inline int hih6130_get_humidity_raw(hih6130_t *dev, uint16_t *humidity_raw)
+{
+    int status;
+    uint8_t buf[HIH6130_HUMIDITY_DATA_LENGTH];
+
+    i2c_acquire(dev->i2c);
+
+    if (i2c_read_bytes(dev->i2c, dev->addr, (char*)&buf[0], sizeof(buf)) != sizeof(buf)) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    i2c_release(dev->i2c);
+
+    /* data is in big-endian format, with status bits in the first byte. */
+    switch (buf[0] & HIH6130_STATUS_MASK) {
+        case HIH6130_STATUS_OK:
+            status = 0;
+            break;
+        case HIH6130_STATUS_STALE_DATA:
+            status = 1;
+            break;
+        default:
+            return -2;
+    }
+
+    *humidity_raw    = ((buf[0] << 8) | buf[1]) & HIH6130_HUMIDITY_MASK;
+
+    return status;
+}
+
+static inline int hih6130_get_humidity_temperature_raw(hih6130_t *dev, uint16_t *humidity_raw, uint16_t *temperature_raw)
+{
+    int status;
+    uint8_t buf[HIH6130_FULL_DATA_LENGTH];
+
+    i2c_acquire(dev->i2c);
+
+    if (i2c_read_bytes(dev->i2c, dev->addr, (char*)&buf[0], sizeof(buf)) != sizeof(buf)) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    i2c_release(dev->i2c);
+
+    /* data is in big-endian format, with status bits in the first byte. */
+    switch (buf[0] & HIH6130_STATUS_MASK) {
+        case HIH6130_STATUS_OK:
+            status = 0;
+            break;
+        case HIH6130_STATUS_STALE_DATA:
+            status = 1;
+            break;
+        default:
+            return -2;
+    }
+
+    *humidity_raw    = ((buf[0] << 8) | buf[1]) & HIH6130_HUMIDITY_MASK;
+    *temperature_raw = (((buf[2] << 8) | buf[3]) >> HIH6130_TEMPERATURE_SHIFT);
+
+    return status;
+}
+
+int hih6130_get_humidity_float(hih6130_t *dev, float *relative_humidity_percent)
+{
+    uint16_t tmp;
+    int status;
+
+    if (hih6130_measurement_request(dev) != 0) {
+        return -1;
+    }
+
+    vtimer_sleep(measurement_delay);
+
+    status = hih6130_get_humidity_raw(dev, &tmp);
+
+    if (status < 0) {
+        return -1;
+    }
+
+    *relative_humidity_percent = tmp * (100.f / 16383.f);
+
+    return status;
+}
+
+int hih6130_get_humidity_temperature_float(hih6130_t *dev,
+    float *relative_humidity_percent, float *temperature_celsius)
+{
+    uint16_t hum_raw, temp_raw;
+    int status;
+
+    if (hih6130_measurement_request(dev) != 0) {
+        return -1;
+    }
+
+    vtimer_sleep(measurement_delay);
+
+    status = hih6130_get_humidity_temperature_raw(dev, &hum_raw, &temp_raw);
+
+    if (status < 0) {
+        return -1;
+    }
+
+    *relative_humidity_percent = hum_raw * (100.f / 16383.f);
+    *temperature_celsius = temp_raw * (165.f / 16383.f) - 40.f;
```

Comment:
the formula is taken directly from the data sheet. the 165 is the span of the sensor (-40 .. +125 C), the -40 is the offset